### ISSUE TITLE
Weight norm

### ISSUE THREAD

**Issue by albertz** [rwth-i6/returnn_common#91]
Implement weight norm.
The implementation can look very similar to PyTorch.

Probably similar would be weight dropout (#100) or other transformations or reparameterizations of weights.

The more generic issue is #59.

**Comment by albertz** [comment]
Note that weight norm only makes sense on model parameters but not on auxiliary parameters (used for e.g. collecting running statistics or so).

**Comment by albertz** [comment]
While implementing weight norm, I stumbled upon the problem of weight decay on its parameterization (#241). Specifically, we would not want weight decay on `g` (or do we?).

In Lingvo, they solved this by parameterizing it as `1 + g` instead ([here](https://github.com/tensorflow/lingvo/blob/65699192ba14521f330a1cae16141433453f1cbf/lingvo/core/layers.py#LL412C25-L412C25)). Do we also want this? Or do it just like layer norm?

**Comment by albertz** [comment]
We now have an initial implementation, see `nn.weight_norm`.
